pkg/controller/nodeconfig/nodeconfig_controller.go

```diff
@@ -0,0 +1,462 @@
+	machineSet, ok := owner.(*clusteroperator.MachineSet)
+	if !ok {
+		c.controller.logger.WithFields(log.Fields{"owner": owner.GetName(), "namespace": owner.GetNamespace()}).
+			Errorf("attempt to enqueue owner that is not a machineset")
+		return
+	}
+	c.controller.enqueueMachineSet(machineSet)
+}
+
+type jobSyncStrategy struct {
+	controller *Controller
+}
+
+func (s *jobSyncStrategy) GetOwner(key string) (metav1.Object, error) {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return nil, err
+	}
+	if len(namespace) == 0 || len(name) == 0 {
+		return nil, fmt.Errorf("invalid key %q: either namespace or name is missing", key)
+	}
+	return s.controller.machineSetsLister.MachineSets(namespace).Get(name)
+}
+
+func (s *jobSyncStrategy) DoesOwnerNeedProcessing(owner metav1.Object) bool {
+	machineSet, ok := owner.(*clusteroperator.MachineSet)
+	if !ok {
+		s.controller.logger.Warn("could not convert owner from JobSync into a machineset: %#v", owner)
+		return false
+	}
+
+	msLog := logging.WithMachineSet(s.controller.logger, machineSet).WithField("func", "DoesOwnerNeedProcessing")
+	if !machineSet.Status.Installed {
+		msLog.Debug("control plane not installed, returning false")
+		return false
+	}
+
+	// Process if the machine set generation has changed since we last ran:
+	if machineSet.Status.NodeConfigInstalledJobGeneration != machineSet.Generation {
+		msLog.WithFields(log.Fields{
+			"NodeConfigInstalledJobGeneration": machineSet.Status.NodeConfigInstalledJobGeneration,
+			"Generation":                       machineSet.Generation,
+		}).Debug("machine set generation has changed")
+		return true
+	}
+	return false
+}
+
+func (s *jobSyncStrategy) DoesOwnerNeedReProcessing(owner metav1.Object) bool {
+	machineSet, ok := owner.(*clusteroperator.MachineSet)
+	if !ok {
+		s.controller.logger.Warn("could not convert owner from JobSync into a machineset: %#v", owner)
+		return false
+	}
+
+	msLog := logging.WithMachineSet(s.controller.logger, machineSet).WithField("func", "DoesOwnerNeedReProcessing")
+
+	// Process if it has been more than the configurable interval of time since we last ran:
+	installingCondition := controller.FindMachineSetCondition(machineSet, clusteroperator.MachineSetNodeConfigInstalling)
+	installedCondition := controller.FindMachineSetCondition(machineSet, clusteroperator.MachineSetNodeConfigInstalled)
+	if installedCondition == nil || installingCondition == nil {
+		msLog.Warn("machine set has node config installed generation but missing relevant conditions")
+		return false
+	}
+
+	if installingCondition.Status == kapi.ConditionTrue {
+		msLog.Warn("node config install already underway")
+		return false
+	}
+
+	if installedCondition.Status == kapi.ConditionFalse {
+		msLog.Warn("node config installed condition was false")
+		return true
+	}
+
+	// Check how long it's been since we last configured the node config daemonset:
+	lastInstalled := installedCondition.LastProbeTime.Time
```

Comment:
I think this is abusing the LastProbeTime field, since it doesn't (or shouldn't) mean that's when you last ran stuff. I would think either of the following approaches would be better:
1) Add another field to the status to indicate when a ConfigureNode was last run
2) Keep track within the controller when stuff last ran

Reply:
Good ideas, I think I'm leaning towards field, I like the idea of us being able to see if on the object when we need to debug something.